release-tools/build.make

```diff
@@ -60,9 +60,18 @@ endif
 # Specific packages can be excluded from each of the tests below by setting the *_FILTER_CMD variables
 # to something like "| grep -v 'github.com/kubernetes-csi/project/pkg/foobar'". See usage below.
 
+BUILD_IN_DOCKER?=false
+DOCKER_BUILDER_WD=/go/src/github.com/kubernetes-csi/$(notdir $(CURDIR))
+DOCKER_BUILDER_IMG?=golang:$(shell grep go: $(CURDIR)/release-tools/travis.yml | cut -d: -f2,2 | tr -d ' ')
+
 build-%:
 	mkdir -p bin
-	CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-X main.version=$(REV) -extldflags "-static"' -o ./bin/$* ./cmd/$*
+ifeq "$(BUILD_IN_DOCKER)" "true"
+	docker run --rm -v $(CURDIR):$(DOCKER_BUILDER_WD):rw -w $(DOCKER_BUILDER_WD) $(DOCKER_BUILDER_IMG) /bin/sh -c " \
+	  CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-X main.version=$(REV) -extldflags -static' -o ./bin/$* ./cmd/$*"
+else
+	CGO_ENABLED=0 GOOS=linux go build -a -ldflags '-X main.version=$(REV) -extldflags -static' -o ./bin/$* ./cmd/$*
```

Comment:
The Go invocation is complex enough that it warrants putting it into a variable (`GO_BUILD` ?) and then  expanding that variable in those two lines.

Reply:
can do, I'd probably add `$(GOARCH)` expansion in that case as well...

Reply:
or you rather mean just the `docker...` call itself?